pkgs/by-name/vc/vcpkg/package.nix

```diff
@@ -32,12 +37,21 @@ stdenvNoCC.mkDerivation (finalAttrs: {
   installPhase = ''
       runHook preInstall
 
-      mkdir -p "$out/bin" "$out/share/vcpkg/scripts/buildsystems"
-      cp --preserve=mode -r ./{docs,ports,triplets,scripts,.vcpkg-root,versions,LICENSE.txt} "$out/share/vcpkg/"
+      mkdir -p "$out/bin" "$out/libexec/vcpkg" "$out/share/vcpkg/scripts/buildsystems"
+      cp --preserve=mode -r ./{.git,docs,ports,triplets,scripts,.vcpkg-root,versions,LICENSE.txt} "$out/share/vcpkg/"
 
       ${lib.optionalString doWrap ''
+        # If you use a builtin-baseline in a vcpkg.json file, then vcpkg will
+        # try to run “git show” on $out/share/vcpkg. By default, Git will fail
+        # if a Git repo isn’t owned by the current user. We create a wrapper
+        # for git here to make sure that the “git show” command succeeds, even
+        # if the user running vcpkg doesn’t own $out/share/vcpkg.
+        makeWrapper "${git}/bin/git" "$out/libexec/vcpkg/git" \
+          --add-flags -c --add-flags safe.directory="$out/share/vcpkg"
+
         makeWrapper "${vcpkg-tool}/bin/vcpkg" "$out/bin/vcpkg" \
-          --set-default VCPKG_ROOT "$out/share/vcpkg"
+          --set-default VCPKG_ROOT "$out/share/vcpkg" \
+          --prefix PATH : "$out/libexec/git"
```

Comment:
Using `--prefix` means the user can use their own `git`, which would remove this change and likely break this feature. Shouldn't we use `--set`?

Reply:
I thought `--prefix` would put it first in path, overriding any other version of git that could be found in the existing environment?

In any case, I suppose `--set` would require putting the rest of the tooling that `vcpkg` depends on here? Maybe except for the tools in `vcpkg-tool`?

Reply:
> I thought `--prefix` would put it first in path, overriding any other version of git that could be found in the existing environment?

I believe that this is how it works. I tried making my `PATH` use a bogus version of Git, and vcpkg still worked fine. Here’s how I did it:

1. Create a folder named `fake-bin`.
2. Inside `fake-bin`, create a file named `git`. Put the following in that file:
    ```bash
    #!/usr/bin/env bash
    false
    ````
3. Run `chmod +x fake-bin/git`.
4. Run `export PATH="$PWD/fake-bin:$PATH`.
5. Make sure that the fake version of Git is being used. Run `git` and then run `echo $?` to make sure that you get exit status 1.
6. Try using the version of `vcpkg` from this pull request. You’ll see that it will handle a builtin-baseline fine.

Reply:
Sorry, it's a confusion on my part, please forget it